modules/planning/tasks/deciders/path_bounds_decider/path_bounds_decider.cc

```diff
@@ -392,15 +392,16 @@ Status PathBoundsDecider::GeneratePullOverPathBound(
          << pull_over_status->position().y() << "] theta["
          << pull_over_status->theta() << "]";
 
+  auto curr_idx = std::get<3>(pull_over_configuration);
   while (static_cast<int>(path_bound->size()) - 1 >
-         std::get<3>(pull_over_configuration) + kNumExtraTailBoundPoint) {
+         curr_idx + kNumExtraTailBoundPoint) {
     path_bound->pop_back();
   }
-  for (int idx = 0; idx < kNumExtraTailBoundPoint; ++idx) {
-    std::get<1>((*path_bound)[path_bound->size() - 1 - idx]) =
-        std::get<1>((*path_bound)[std::get<3>(pull_over_configuration)]);
-    std::get<2>((*path_bound)[path_bound->size() - 1 - idx]) =
-        std::get<2>((*path_bound)[std::get<3>(pull_over_configuration)]);
+  for (int idx = curr_idx + 1;
+       idx < std::min(static_cast<int>(path_bound->size()),
+                      curr_idx + kNumExtraTailBoundPoint + 1); ++idx) {
+    std::get<1>((*path_bound)[idx]) = std::get<1>((*path_bound)[curr_idx]);
+    std::get<2>((*path_bound)[idx]) = std::get<2>((*path_bound)[curr_idx]);
   }
```

Comment:
you are not really changing anything, right?

Reply:
@jmtao 
Here, I have fixed up the corner case when `path_bound->size() < kNumExtraTailBoundPoint` for runtime error and when `curr_idx + kNumExtraTailBoundPoint > path_bound->size()` for logic error.